rmw_fastrtps_shared_cpp/src/rmw_init.cpp

```diff
@@ -71,25 +64,19 @@ rmw_init_options_copy(
   const rcutils_allocator_t * allocator = &src->allocator;
   rmw_ret_t ret = RMW_RET_OK;
 
+  allocator->deallocate(dst->name, allocator->state);
   *dst = *src;
   dst->name = NULL;
-  dst->namespace_ = NULL;
   dst->security_options = rmw_get_zero_initialized_security_options();
 
   dst->name = rcutils_strdup(src->name, *allocator);
```

Comment:
@ivanpauno nit^2: consider attempting all allocations before actually writing `dst` so that an error leaves does not leave `dst` in a bad state.